python/pyarrow/_dataset.pyx

```diff
@@ -861,10 +861,15 @@ cdef class RowGroupInfo:
             name = frombytes(c_statistics.type.get().field(i).get().name())
             c_minmax = <CStructScalar*> c_statistics.value[i].get()
 
-            statistics[name] = {
-                'min': pyarrow_wrap_scalar(c_minmax.value[0]).as_py(),
-                'max': pyarrow_wrap_scalar(c_minmax.value[1]).as_py(),
-            }
+            try:
+                statistics[name] = {
+                    'min': pyarrow_wrap_scalar(c_minmax.value[0]).as_py(),
+                    'max': pyarrow_wrap_scalar(c_minmax.value[1]).as_py(),
+                }
+            except ValueError:
+                # Don't threat failure to parse/convert a single Scalar as a
```

Comment:
```suggestion
                # Don't treat failure to parse/convert a single Scalar as a
```

Reply:
Do you have examples where you still got an error?

Reply:
Haven't tried, but if a new type id is added an the scalar conversion in python isn't added it'll raise this.